### ISSUE TITLE
Incorrect lap distance?

### ISSUE THREAD

**Issue by AndreiPashkin** [chrisfosterelli/tcxdist#3]
Shouldn't the value here be divided by the number of laps?
https://github.com/chrisfosterelli/tcxdist/blob/8b909412595af77d9800b32bf9c1a85b495fe119/tcxdist.py#L38

**Comment by chrisfosterelli** [comment]
Ah yeah you're right! I don't typically use laps on the trainer so I never looked closely, but when you do them they end up like this:

![image](https://user-images.githubusercontent.com/2072672/103960247-85b7c900-5106-11eb-9de3-ccd490841770.png)

It probably shouldn't be divided just by the number of laps, or else each lap would end up the same size regardless of when the laps were triggered, but proportionally assigned based on the portion of the lap's `TotalTimeSeconds` relative to the full activity length, like [here](https://github.com/chrisfosterelli/tcxdist/blob/8b909412595af77d9800b32bf9c1a85b495fe119/tcxdist.py#L44).

That way you'd get the proper behaviour compared to the trackpoints, where it looks like you held a consistent pace even if you use manual laps.

**Comment by chrisfosterelli** [comment]
Also it's using `args.km` instead of `distance` so it's off by a factor of 1000 too

**Comment by AndreiPashkin** [comment]
@chrisfosterelli, you are right.

**Comment by chrisfosterelli** [comment]
I did a quick ride yesterday and played with a few laps to get some test data so I can fix this soon

**Comment by chrisfosterelli** [comment]
This is now resolved! I had to make some other small tweaks but laps now show as would be expected based on their length in Strava 
![image](https://user-images.githubusercontent.com/2072672/104080163-4c04c200-51db-11eb-98c7-dbedb0733cb4.png)

**Comment by chrisfosterelli** [comment]
Thanks for reporting this @AndreiPashkin

**Comment by AndreiPashkin** [comment]
@chrisfosterelli, thank you for writing this awesome script, I'm using it too now. I wonder why have you decided to use TCX and not GPX - there is a special library for it:
https://github.com/tkrajina/gpxpy

Have you considered it? If yes - can you say something about it?

**Comment by chrisfosterelli** [comment]
@AndreiPashkin I'm glad you find it useful!

I am not an expert on the file formats and didn't investigate thoroughly, but chose TCX for a few reasons. GPX is used to represent series of GPS points, and isn't really a "workout file format", but there are extensions that make it include things like heart rate, etc. for each point, while TCX and FIT are "workout file formats" that are specifically orientated to workouts.

If we used GPX, the biggest problem is it opens up a question of "what should the GPS points be?" We would need to provide latitude and longitude for each trackpoint to be valid GPX. TCX trackpoints _can_ have lat / lon but aren't _required_ to have them. Trainer data doesn't inherently have GPS coordinates, and if we used just a fixed location it wouldn't have any speed or distance metrics in the resulting file. With TCX we can embed these values directly.

FIT would have also worked but it's a binary format and the XML format of TCX is a little easier to work with.

Did you have a specific need for GPX?
